traffic_monitor/tests/gocover.bash

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#----------------------------------------
+set -e
+go get "github.com/wadey/gocovmerge"
+echo "" > result.txt
+packages=( "$@" )
+coverage_out=()
+i=1
+for pkg in ${packages[@]} ; do
+    for d in $(go list $pkg | grep -v vendor); do
+        file="$i.out"
+        go test -v -coverprofile=$file $d >> result.txt
+        cat result.txt
```

Comment:
So now, for every iteration of each loop you're printing the entire results file, which means e.g. you're going to see the first line again on every iteration. Correct me if I'm wrong, but I think what would be better is to print out each line as you go. To do that, you can use [`tee(1)`](http://man7.org/linux/man-pages/man1/tee.1.html) like so:

```bash
go test -v --coverprofile "$file" "$d" | tee -a result.txt # note the quotes around shell variables to prevent interpretation
```